lib/dialogs/buy/buy_dialog.dart

```diff
@@ -278,7 +278,14 @@ class _BuyDialogState extends State<BuyDialog> {
       controller.publicAddress.toString(),
     );
 
-    return BlocBuilder<BuyDialogBloc, BuyDialogState>(
+    return BlocConsumer<BuyDialogBloc, BuyDialogState>(
+      listenWhen: (previous, current) =>
+          previous.status != BlocStatus.error &&
+          current.status == BlocStatus.error,
+      listener: (context, state) => context.showCommonWarningToast(
+        title: 'Action Error',
+        desc: 'An error has occured while handling the Buy action',
+      ),
       buildWhen: (previous, current) => previous != current,
```

Comment:
this `buildWhen` can be removed, it literally does nothing. it's what the underneath bloc logic is by default.